database/src/resource-repository.ts

```diff
@@ -55,6 +56,36 @@ export default class ResourceRepository implements IRepository<IResource> {
       }),
       resultOfSelectRepresentations)
   }
+  async * iterateSpaceRepresentationsWithLinks(query: { space: string }) {
+    const rows = await this.#database
+      .selectFrom('resourceRepresentation')
+      .innerJoin('spaceNamedResource', 'spaceNamedResource.resourceId', 'resourceRepresentation.resourceId')
+      .innerJoin('blob', 'blob.uuid', 'resourceRepresentation.representationId')
+      .leftJoin('link', 'link.anchor', 'resourceRepresentation.resourceId')
+      .select([
+        'blob.bytes',
+        'blob.type',
+        'resourceRepresentation.createdAt',
+        'spaceNamedResource.spaceId as space',
+        'resourceRepresentation.resourceId as resourceId',
+        'link.anchor as linkAnchor',
+        'link.uuid as linkId',
+      ])
+      .where('spaceNamedResource.spaceId', '=', query.space)
+      .orderBy('resourceRepresentation.createdAt', 'desc')
+      .execute()  
+      for (const x of rows) {
+        const base = x.linkAnchor || x.linkId || x.resourceId
+        const fileName = `${base}?ct=${x.type}.${extension(x.type)}`.replace(/[\/\\:*"<>|]/g, '_')
```

Comment:
```suggestion
        const fileName = `${base}?ct=${x.type}.${extension(x.type)}`.replace(/[\/\\:*"<>|]/g, '_')
```
Extract this logic to a separate function (so that the generation of filenames can be unit tested separately).
Also, `replace(/[\/\\:*"<>|]/g, '_')` might not work for all reserved characters on all operating systems.
Instead, I recommend percent-encoding (which is typically used for making URLs safe from reserved characters, but it ALSO is useful for files).
So, something like:
```
encodeURIComponent(`${base}?ct=${x.type}`) + `${extension(x.type)}`
```